server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts

```diff
@@ -180,8 +182,35 @@ export class DeliLambdaFactory extends EventEmitter implements IPartitionLambdaF
 					Lumberjack.info(message, getLumberBaseProperties(documentId, tenantId));
 				}
 			} else {
-				lastCheckpoint = JSON.parse(document.deli);
-			}
+                let checkpoint;
+                let isLocalCheckpoint = false;
+                if(this.localCheckpointEnabled){
+                    // Search local db for checkpoint
+                    Lumberjack.info(`Searching local collection for checkpoint.`, getLumberBaseProperties(documentId, tenantId));
+                    checkpoint = await this.localCheckpointCollection.findOne( {documentId, tenantId }).catch((error) => {
+                        Lumberjack.error(`Error retrieving local checkpoint`, getLumberBaseProperties(documentId, tenantId));
+                    });
+
+                    if(checkpoint !== null && checkpoint !== undefined){
+                        lastCheckpoint = JSON.parse(checkpoint.deli);
+                        isLocalCheckpoint = true;
+                    } else {
+                        lastCheckpoint = JSON.parse(document.deli);
+                    }
+
+                } else {
+                    lastCheckpoint = JSON.parse(document.deli);
+                }
+
+                const lumberjackProperties = {
```

Comment:
can we also log the checkpointTimestamp here? Might be useful to check the effectiveness of our checkpointing heuristics